### ISSUE TITLE
Root project not yet available if subproject listed in settings.gradle is missing.

### ISSUE THREAD

**Issue by vertu20140207** [gradle/gradle#1733]
### Expected Behavior
I apply an initialization script that uses settingsEvaluated and it.rootProject against a project whose settings.gradle includes a project that doesn't exist. Gradle should either ignore the missing sub-project (since the build itself still works in this case), or it should provide a more informative error message than the current "The root project is not yet available for build.", i.e. "The root project is not yet available for build due to a missing sub-project <subproject name>.".

### Current Behavior
None of --info, --debug and --stacktrace are useful in identifying the root cause of this problem and the current error message is not very informative, so it's hard to debug this problem without understanding the gradle lifecycle.
"* What went wrong:
The root project is not yet available for build."

### Context
I'm using some generic initialization scripts to modify third-party builds so I can run static checkers, re-sign output etc. without needing to modify the original build files which I don't own.

### Steps to Reproduce (for bugs)
In this project: https://github.com/microg/android_packages_apps_GmsCore, apply the signing init script from this project: https://github.com/vertu20140207/gradle-init-scripts. i.e. ./gradlew -I <path>/signing-init.groovy clean
Remove the first line from settings.gradle and the problem goes away.
It's probably not hard to generate a reduced toy example.

### Your Environment
Not really relevant, I've explained the root cause.

**Comment by eriwen** [comment]
A couple things would help us consider this:
 * Gradle version or a build scan
 * A simpler example

As it stands now, I think it'll take a lot of time for us to break this down into a simple example. As someone more familiar with the code, would you please consider this?

**Comment by vertu20140207** [comment]
OK, it's not what I originally thought, so the title is misleading/wrong. The reduced example is that this init script doesn't work in a multi-project build:
~~~~
settingsEvaluated { settings ->
    settings.rootProject.children.each {
        rootProject.logger.log(LogLevel.INFO, it.toString())
    }
}
~~~~
The rootProject I want to get a logger from isn't yet available. In this case, I see I should just be using logger.info(), so it's no big deal. Is it documented somewhere when rootProject will be available?

**Comment by eriwen** [comment]
@vertu20140207 Thanks for the update. 

I'm not sure this is documented, but tracing the code the `rootProject` is set after init scripts are executed in [DefaultGradleLauncher](https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncher.java#L135). `findAndLoadSettings()` eventually calls `setRootProject`. 

TL;DR: the root project isn't available in init scripts.
